lib/routes/routeBackbeat.js

```diff
@@ -472,14 +472,29 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
                 omVal[headerName] = objMd[headerName];
             });
         }
+
         const versionId = decodeVersionId(request.query);
+        const versioning = bucketInfo.isVersioningEnabled();
+
         // specify both 'versioning' and 'versionId' to create a "new"
         // version (updating master as well) but with specified
         // versionId
         const options = {
-            versioning: bucketInfo.isVersioningEnabled(),
+            versioning,
             versionId,
         };
+
+        if (versionId === 'null') {
+            // Retrieve the null version id from the object metadata.
+            const nullVersionId = objMd && objMd.versionId;
+            if (versioning && !nullVersionId) {
+                // If the null version does not have a version id, it is a current null version.
+                // To update the metadata of a current version, versioning is set to false.
+                options.versioning = false;
```

Comment:
Beware of this, as option fields are converted to strings when passed to Metadata (through the query string), there's a good chance that we interpret `versioning: "false"` as being true in Metadata, I'd rather recommend to not pass `versioning` at all if it is false. Metadata should have better parsing of its options on the other hand but I don't think it's done properly now.